File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/PostTeamExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class PostTeamExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey.Add("Authorization", "YOUR_API_KEY");

        var teamPostInput = new TeamPostInput(
            key: "team-key-123abc",
            name: "Example team",
            description: "An example team",
            customRoleKeys: [
                "example-role1",
                "example-role2",
            ],
            memberIDs: [
                "12ab3c45de678910fgh12345",
            ]
        );

        try
        {
            var response = new TeamsApi(config).PostTeam(
                teamPostInput: teamPostInput
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling TeamsApi#PostTeam: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/PostTeamExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const teamPostInput: api.TeamPostInput = {
  key: "team-key-123abc",
  name: "Example team",
  description: "An example team",
  customRoleKeys: [
    "example-role1",
    "example-role2",
  ],
  memberIDs: [
    "12ab3c45de678910fgh12345",
  ],
};

new api.TeamsApi(configuration).postTeam(
  teamPostInput,
  undefined, // expand
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling TeamsApi#postTeam:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
